ui/app/src/components/MediaCard.tsx

```diff
@@ -0,0 +1,257 @@
+    paddingTop: '56.25%',
+    position: 'relative',
+    '& .media-icon': {
+      position: 'absolute',
+      top: '50%',
+      left: '50%',
+      transform: 'translate(-50%, -50%)',
+      color: 'rgba(0, 0, 0, 0.54)',
+      fontSize: '50px'
+    },
+    '&.list': {
+      height: '80px',
+      width: '80px',
+      paddingTop: '0',
+      order: -1
+    }
+  },
+  deleteIcon: {
+    marginLeft: 'auto',
+  },
+  mLa: {
+    marginLeft: 'auto'
+  },
+  checkbox: {
+    marginLeft: '28px',
+    '& label': {
+      marginRight: 0
+    },
+    '&.list': {
+      justifyContent: 'center',
+      order: -2,
+      marginRight: '16px'
+    }
+  },
+}));
+
+interface MediaCardProps {
+  item: MediaItem;
+  currentView: string;
+  selected: Array<string>;
+
+  handleEdit(path: string): any;
+
+  handleDelete(path: string): any;
+
+  handlePreview(url: string): any;
+
+  handlePreviewAsset(url: string, type: string, name: string): any;
+
+  handleSelect(path: string, selected: boolean): void;
+}
+
+function MediaCard(props: MediaCardProps) {
+  const classes = useStyles({});
+
+  const {handleEdit, handleDelete, handlePreview, handlePreviewAsset, handleSelect, selected, item} = props;
+  const {name, path, type} = item;
+  const isList = props.currentView === 'list';
+  const siteUrl = 'http://localhost:8080';
+
+  function renderIcon(type: string, path: string) {
+    let iconClass = 'fa media-icon';
+    let iconName = `${iconClass} fa-file`;
+    let actionArea = false;
+    switch (type) {
+      case 'Page':
+        iconName = `${iconClass} fa-file`;
+        actionArea = true;
+        break;
+      case 'Video':
+        iconName = `${iconClass} fa-file-video-o`;
+        break;
+      case 'Template':
+        iconName = `${iconClass} fa-file-code-o`;
+        break;
+      case 'Taxonomy':
+        iconName = `${iconClass} fa-tag`;
+        break;
+      case 'Component':
+        iconName = `${iconClass} fa-puzzle-piece`;
+        break;
+      case 'Groovy':
+        iconName = `${iconClass} fa-file-code-o`;
+        break;
+      default:
+        break;
+    }
+    return (
+      actionArea ?
+        <CardActionArea onClick={() => handlePreview(path)} className={clsx(isList && classes.listActionArea)}>
+          <div className={clsx(classes.mediaIcon, isList && 'list')}>
+            <i className={iconName}></i>
+          </div>
+        </CardActionArea>
+        :
+        <div className={clsx(classes.mediaIcon, isList && 'list')}>
+          <i className={iconName}></i>
+        </div>
+    )
+  }
+
+  const handleDetailsClick = (path: string, type: string, name: string) => {
+    if (type === 'Image' || type === 'Video') {
+      handlePreviewAsset(path, type, name);
+    } else {
+      handleEdit(path);
+    }
+  };
+
+  return (
+    <Card className={clsx(classes.card, isList && 'list')}>
+      {
+        isList &&
+        <FormGroup className={clsx(classes.checkbox, 'list')}>
+          <FormControlLabel
```

Comment:
If it doesn't have a label, why not directly use the `Checkbox` control? 

Add for/aria attributes for control/label correspondence.

Reply:
Unresolved